Assets/Grid.cs

```diff
@@ -1,91 +1,160 @@
+using System;
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 using CodeMonkey.Utils;
-
 /// <summary>
 /// Responsible for spawning grids where neccessary
 /// </summary>
-public class Grid 
+public class BattleGrid
 {
+
+    public const int HEAT_MAP_MAX_VALUE = 100;
+    public const int HEAT_MAP_MIN_VALUE = 0;
+
+    public event EventHandler<OnGridValueChangedEventArgs> OnGridValueChanged;
+    public class OnGridValueChangedEventArgs : EventArgs
+    {
+        public int x;
+        public int y;
+    }
+
     private int width;
     private int height;
     private float cellSize;
     private Vector3 originPosition;
     private int[,] gridArray;
-    private TextMesh[,] debugTextArray;
 
-    public Grid (int width, int height, float cellSize, Vector3 originPosition)
+    public BattleGrid(int width, int height, float cellSize, Vector3 originPosition)
     {
         this.width = width;
         this.height = height;
         this.cellSize = cellSize;
         this.originPosition = originPosition;
 
         gridArray = new int[width, height];
-        debugTextArray = new TextMesh[width, height];
 
-        //Draw the grid based on values
-        for (int x= 0; x < gridArray.GetLength(0); x++)
+        bool showDebug = true;
+        if (showDebug)
         {
-            for (int y = 0; y < gridArray.GetLength(1); y++)
+            TextMesh[,] debugTextArray = new TextMesh[width, height];
+
+            for (int x = 0; x < gridArray.GetLength(0); x++)
             {
-                debugTextArray[x, y] = UtilsClass.CreateWorldText(gridArray[x, y].ToString(), null, GetWorldPosition(x, y) + new Vector3(cellSize, cellSize) *.5f, 20, Color.white, TextAnchor.MiddleCenter);
-                Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x, y + 1), Color.white, 100f);
-                Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x + 1, y), Color.white, 100f);
+                for (int y = 0; y < gridArray.GetLength(1); y++)
+                {
+                    debugTextArray[x, y] = UtilsClass.CreateWorldText(gridArray[x, y].ToString(), null, GetWorldPosition(x, y) + new Vector3(cellSize, cellSize) * .5f, 30, Color.white, TextAnchor.MiddleCenter);
+                    Debug.DrawLine(GetWorldPosition(x, y), GetWorldPosition(x, y + 1), Color.white, 100f);
```

Comment:
Are there magic numbers in lines 46-56
46: new Vector3 - .5f, 30
47-56; 100f, x + 1